### PR TITLE
Download workshop data for the GitHub Action, and move real vignetted in its rightful place

### PR DESCRIPTION
**Pull request by kevinrue**
Don't merge just yet. I need to pull in your recent changes to `master` and cleanup a couple of things first.

### CODE REVIEW

**Comment by rorystark** [comment]
Thanks Kevin, let me know when I should merge your fixes.

-R

On 7 Dec 2020, at 11:46, Kevin Rue-Albrecht <notifications@github.com> wrote:

﻿

Don't merge just yet. I need to pull in your recent changes to master and cleanup a couple of things first.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/bioinformatics-core-shared-training/Quantitative-ChIPseq-Workshop/pull/2

Commit Summary

  *   try GHA to download workshop data
  *   use actions-download-file instead
  *   just use curl
  *   dont use an Rscript ...
  *   dont refer to full paths specific to docker
  *   Merge branch 'main' into kra-gha
  *   reactivate basic checks; bring vignette into its rightful place

File Changes

  *   M .github/workflows/basic_checks.yaml<https://github.com/bioinformatics-core-shared-training/Quantitative-ChIPseq-Workshop/pull/2/files#diff-552711de6b7d8e1956debace1c0b49cb4e9b20d799f1143274d87be5f69737a1> (40)
  *   M vignettes/Quantitative-ChIPseq-Workshop.Rmd<https://github.com/bioinformatics-core-shared-training/Quantitative-ChIPseq-Workshop/pull/2/files#diff-171502bb389b73dfe1aaa36f4ce8ed703be973e77e8070d7adf814737e40e795> (600)

Patch Links:

  *   https://github.com/bioinformatics-core-shared-training/Quantitative-ChIPseq-Workshop/pull/2.patch
  *   https://github.com/bioinformatics-core-shared-training/Quantitative-ChIPseq-Workshop/pull/2.diff

**Comment by kevinrue** [comment]
You can merge the PR.

Everything went well until this line https://github.com/kevinrue/Quantitative-ChIPseq-Workshop/runs/1510480358?check_suite_focus=true#step:14:26760

After that, as expected, the GitHub Action on my account did not have the credentials to push the successfully built image to your DockerHub account.

**Comment by rorystark** [comment]
Merging Kevin's fixes.

**Comment by rorystark** [comment]
It has all built and the Docker image pushed, awesome!

Nor I'm un-doing my slight-of-hand and removing inst/vignettes so vignettes will be handled in the correct way. Hopefully that all builds fine (I commented out the line in the Dockerfile that copied the vignette to vignettes/). The build is running now.

I do have two questions about how to actually run the workshop:

  *   I notice when I run browseVignettes() in the RStudio window (after explicitly loading the QuantitativeChIPseqWorkshop  library), it shows the QuantitativeChIPseqWorkshop vignette, but the links are all broken. They point to http://localhost:8787/library/QuantitativeChIPseqWorkshop/doc/Quantitative-ChIPseq-Workshop.html and the files are actually in /usr/local/lib/R/site-library/QuantitativeChIPseqWorkshop/doc/. How should users load the .html vignette?
  *   When running the workshop, it is customary to have the attendees be loking at:
     *   The html vignette? In another tab perhaps?
     *   The vignette .Rmd source in the RSdution Source pane, so they can run the code chunks as we go?
     *   Something else? Live coding?

Thanks again for all the effort you've put into this!

Cheers-
R

________________________________

**Comment by kevinrue** [comment]
Your first point is a known issue in Docker containers.

See comments in this section: https://isee.github.io/iSEEWorkshopEuroBioc2020/#workshop-setup
Briefly, you can either add `help/` to the URL, or you can instead use `vignette('overview', 'iSEEWorkshopEuroBioc2020')` as we do here: https://isee.github.io/iSEEWorkshopEuroBioc2020Slides/#9

For your second point, I would encourage participants to use the vignette compiled by your GHA and deployed at https://bioinformatics-core-shared-training.github.io/Quantitative-ChIPseq-Workshop/articles/Quantitative-ChIPseq-Workshop.html
And as you say, they can then have it open in another tab.
The bonus point there you'll take them straight to the repository that they are then more likely to star for future reference. Otherwise they may forget to do it later, or worse, not even be aware that the repo exists.

Then it's up to your style of workshop. The Rmd vignette is only really useful to participants if they run the code themselves. This happens only for workshops that give people periods of time when the speaker stops talking and lets people work through a part of the vignette or an exercise. Workshops that are less interactive and more of a demonstration led by the instructor generally don't get as much interactive engagement from the participants with the vignette; instead they tend to just watch what happens on the screen of the speaker.

In summary, and partially covering your last point, the delivery of the workshop is very much up to you and how you would like to engage your participants with the workshop materials.
Your workshop was accepted as a "Long workshop (1.5 hours) in an interactive format, where participants will be expected to perform an analysis themselves".
From experience, for yourself as well as for your participants, I would plan some of that time to give participants a chance to run parts of the vignette themselves. That will also give you time to take a breath and hydrate while assessing where you are in the workshop and planning for the remaining time.

I hope that helps!
